### DISCUSSION TITLE
​New Feature Request: Folder Mode

### DISCUSSION THREAD

**Discussion by houshasen** [toeverything/AFFiNE#6988] [Ideas]
### Background
This concept is inspired by two main factors:

**Familiarity with Folder Structures**: We are accustomed to using folder structures on computers daily. This habitual use influences how we organize and conceptualize information, making folder structures a natural choice for many of us.

**Heptabase's Nested Whiteboard**: The second source of inspiration is Heptabase's nested whiteboard, which functions similarly to a hierarchical folder system. This approach effectively mirrors the organizational benefits of traditional folder structures.

### Request

Please consider creating a Folder mode like Page, Edgeless Mode. This gives true visual folder organization view, but in a AFFiNE way.

### Detail 
**Folder Pane**

- The primary purpose of this view is to facilitate easy navigation through linked documents, whether they are networked or hierarchical.
- Each linked document will be represented by a folder-equivalent entity.
- Notes at the current level will be displayed as note entities.
- Therefore, when switching to folder mode in any document, users can quickly see the number and types of notes present, as well as the documents being referenced on it.

**Folder entity**

- Instead of merely displaying a folder icon with the name of each linked document, we should provide additional information that helps user to navigate. For example, we could not only shows the linked document name but also includes notes, and linked doc names within the folder entity, similar to Heptabase.

**Note entity**

- Firstly, we can utilize the Grouping feature created on Edgeless to display grouped notes here as well.
- The amount and manner of displaying note content is an interesting design decision. For instance, we could include a description property and each look like index card (like Scrivener),  perhaps show the first couple of paragraphs or solely just Note title alone.

**Other files**
Other blocks can also be shown grouped here e.g. PDF block, Media block etc.

**Note View**

To complete the use case, we need a way to interact with the notes. Therefore, each note entity should have a "maximize" icon. Clicking on this icon will bring the note into full view or a part of the pane, allowing us to fully edit it.

### Use case

The real-world use case for this feature is akin to a desktop folder structure, enabling much quicker access to specific notes, especially when managing a large number of them.

For instance, I currently have over 10 notes in one document, each containing multiple paragraphs. Additionally, I may have some media files etc. In Edgeless mode, these elements are spread out. If I want to view all of them and set the screen to fit the size, I can't read anything. If I zoom in to a readable level, I have to scroll around. With Folder Mode, everything will be on one screen. I can simply choose a note I want and start editing.
It also offers those who want real folder structure to experience that.

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by EYHN** [comment]
@hwangdev97 take a look

**Comment by hwangdev97** [comment]
We are working for this. Maybe you can use it on version 0.17
![shot-2024-05-20-21 29 49@2x](https://github.com/toeverything/AFFiNE/assets/24713927/bfc39434-b686-4355-8e9e-10e5fd51d272)

**Reply by CX330Blake** [reply]
Hi can I ask how did you made this beautiful screen shot?

**Comment by houshasen** [comment]
> We are working for this. Maybe you can use it on version 0.17 

Wow, this must be a spoiler. I'm trying to decipher this.

It now says "Organize" instead of "Collection," and we have a "Collection A" node. I assume you are working on expanding organization beyond just collections.

I also see a folder icon, which we don't have right now, that contains "Collection A." So, I'm guessing it's a real folder where we can drop anything, including collections and documents. Can any of those points show "notes" or other items e.g. PDF block, Media block within the given document?

By the way, version 0.17 sounds like it's going to be big with lots of new features.

**Reply by hwangdev97** [reply]
We hope that a folder is an entity used solely for organizing content, and it can be used to store:

- collections
- docs
- tags
- subfolders

Of course, you can also favorite a folder. 

However, the specific details have not been fully determined yet.

**Reply by hwangdev97** [reply]
We do not want a PDF or other attachment block to appear as a standalone file entity in the product;

 it still needs a doc to contain it. Therefore, a folder will not be able to store a PDF block or a Media block.

**Reply by houshasen** [reply]
Interesting. So, are PDF and Media blocks considered pure files? I initially thought the term "block" implied they came with a wrapper. Currently, I understand that PDFs or Media files in Edgeless mode are uploaded without a wrapper. Could you explain why this is the case and why you don't always wrap them?

I can see the benefits of wrapping files, such as the allowing them to take notes, potentially assign tags, and even properties. However, I'm unsure about the advantages of having unwrapped files, especially if you don't want them to be accessed as standalone file entities.

It's also interesting that the team is considering a regular "folder" organization. Many Personal Knowledge Management systems tend to avoid this, emphasizing and sometimes forcing a "network-note taking" approach. Personally, I wouldn't have any issues if we adopted a regular folder organization. It offers flexibility to users. We just need to ensure that we understand when to use folders, tags and collection.
